src/workerd/api/sockets.c++

```diff
@@ -262,7 +263,7 @@ jsg::Ref<Socket> connectImpl(
   return connectImplNoOutputLock(js, kj::mv(fetcher), kj::mv(address), kj::mv(options));
 }
 
-jsg::Promise<void> Socket::close(jsg::Lock& js) {
+jsg::Promise<void> Socket::closeImplOld(jsg::Lock& js) {
```

Comment:
Some comments somewhere in here explaining the differences between these two closeImpl variants would be good.